google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/client/BigQueryCommunicationHelper.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.bigquery.client
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, HttpMethods, HttpRequest}
+import akka.stream.alpakka.googlecloud.bigquery.client.QueryJsonProtocol.{QueryRequest, QueryResponse}
+import akka.stream.alpakka.googlecloud.bigquery.client.TableDataQueryJsonProtocol.TableDataQueryResponse
+import akka.stream.alpakka.googlecloud.bigquery.client.TableListQueryJsonProtocol.TableListQueryResponse
+import spray.json.JsObject
+
+import scala.util.Try
+
+object BigQueryCommunicationHelper {
+
+  def createQueryRequest(query: String, projectId: String, dryRun: Boolean) =
+    HttpRequest(HttpMethods.POST, GoogleEndpoints.queryUrl(projectId), entity = createQueryBody(query, dryRun))
+
+  def createQueryBody(query: String, dryRun: Boolean) =
+    HttpEntity(ContentTypes.`application/json`, QueryRequest(query, dryRun = Some(dryRun)).toJson.compactPrint)
+
+  def parseQueryResult(result: JsObject): Try[(Seq[String], Seq[Seq[String]])] =
```

Comment:
This signature (and the following methods) is very Scala-specific. Its use is just internal, they should move somewhere else.

Reply:
They were set as an bigquery package private. Examples using it have been adjusted
`InternalApi
private[bigquery]`